From 97c3d7585e5a88ab2aa8d69fc4e3b6b624dd8b84 Mon Sep 17 00:00:00 2001
From: Bai Ping <b51503@freescale.com>
Date: Wed, 1 Jul 2015 01:19:14 +0800
Subject: [PATCH 0416/5242] MLK-11204 ARM: imx: set int_mem_clk_lpm bit bit
 when entering wait mode on 6ul

commit  7cdadc3a72b57b4de1b0095073e1cf471fd3f4b7 from
https://source.codeaurora.org/external/imx/linux-imx.git

This bit is used to keep the ARM Platform memory clocks enabled if
an interrupt is pending when entering low power mode. This bit should
always bet set when the CCM_CLPCR_LPM bits are set to 01(WAIT Mode) or
10 (STOP mode) without power gating.

Signed-off-by: Bai Ping <b51503@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/mach-imx/cpuidle-imx6ul.c |    3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm/mach-imx/cpuidle-imx6ul.c b/arch/arm/mach-imx/cpuidle-imx6ul.c
index c80e034..03a7e1b 100644
--- a/arch/arm/mach-imx/cpuidle-imx6ul.c
+++ b/arch/arm/mach-imx/cpuidle-imx6ul.c
@@ -178,5 +178,8 @@ int __init imx6ul_cpuidle_init(void)
 	imx6ul_wfi_in_iram_fn = (void *)fncpy(wfi_iram_base + sizeof(*cpuidle_pm_info),
 		&imx6ul_low_power_idle, wfi_code_size);
 #endif
+
+	imx6_set_int_mem_clk_lpm(true);
+
 	return cpuidle_register(&imx6ul_cpuidle_driver, NULL);
 }
-- 
1.7.9.5

